src/vs/workbench/contrib/search/browser/searchActionsTextQuickAccess.ts

```diff
@@ -29,6 +34,28 @@ registerAction2(class TextSearchQuickAccessAction extends Action2 {
 
 	override async run(accessor: ServicesAccessor, match: RenderableMatch | undefined): Promise<any> {
 		const quickInputService = accessor.get(IQuickInputService);
-		quickInputService.quickAccess.show(TEXT_SEARCH_QUICK_ACCESS_PREFIX);
+		const searchText = getSearchText(accessor) ?? '';
```

Comment:
Can the input text be configured in the `defaultFilterValue` function instead? https://github.com/microsoft/vscode/blob/e34a26097d763548bf220da77b74922765c12fa2/src/vs/workbench/contrib/search/browser/quickTextSearch/textSearchQuickAccess.ts#L103

I'd like `this.configuration.preserveInput` to only use the last input value only if there is no selection (or we aren't using the selection). Currently, it prioritizes the value from enabling `this.configuration.preserveInput`. I also think that the `defaultFilterValue` function is the proper place to put an initial value.

Reply:
I'm sorry I'm a little confused, I'm not sure if I understand the right information, but the latest change doesn't get the selected content when `preserve input` is turned on, do I need to do some configuration in other configurations, or the way I handled it is wrong

I didn't get the last input here, I think something I got wrong
https://github.com/microsoft/vscode/blob/41e940f76f5deda197bc5930b044c55607ba1cbc/src/vs/platform/quickinput/browser/quickAccess.ts#L82

Reply:
Ah I totally forgot that the `defaultFilterValue` function returns an enum, not the actual last value! In that case, what we had when I made the above comment should be okay. Sorry for the confusion 😅

Reply:
> Ah I totally forgot that the `defaultFilterValue` function returns an enum, not the actual last value! In that case, what we had when I made the above comment should be okay. Sorry for the confusion 😅

Haha then I'll revert the code back😂, I saw that the place where this method was called would also determine whether the value obtained was a string, and I thought I could return the string as well